components/cam/src/physics/cam/cam_diagnostics.F90

```diff
@@ -1681,18 +1683,64 @@ subroutine diag_surf (cam_in, cam_out, ps, trefmxav, trefmnav )
     integer :: ncol         ! longitude dimension
     real(r8) tem2(pcols)    ! temporary workspace
     real(r8) ftem(pcols)    ! temporary workspace
+#ifdef MAML
+    real(r8) shfavg(pcols),lhfavg(pcols),wsxavg(pcols),wsyavg(pcols)
+    real(r8) snowhlandavg(pcols),asdiravg(pcols),aldiravg(pcols)
+    real(r8) asdifavg(pcols),aldifavg(pcols)
+    real(r8) lhfsd(pcols)
+    integer :: ii
+#endif
 !
 !-----------------------------------------------------------------------
 !
     lchnk = cam_in%lchnk
     ncol  = cam_in%ncol
-
+#ifdef MAML
+    lhfavg = 0.; shfavg = 0.; wsxavg = 0.; wsyavg = 0.
+    snowhlandavg = 0.; asdiravg = 0.; aldiravg = 0.
+    asdifavg = 0.; aldifavg = 0.
+    lhfsd = 0.
+    do i = 1,ncol
+      do ii = 1,num_inst_atm
+        lhfavg(i) = lhfavg(i)+cam_in%lhf(i,ii)
+        shfavg(i) = shfavg(i)+cam_in%shf(i,ii)
+        wsxavg(i) = wsxavg(i)+cam_in%wsx(i,ii)
+        wsyavg(i) = wsyavg(i)+cam_in%wsy(i,ii)
+        snowhlandavg(i) = snowhlandavg(i)+cam_in%snowhland(i,ii)
+        asdiravg(i) = asdiravg(i)+cam_in%asdir(i,ii)
+        aldiravg(i) = aldiravg(i)+cam_in%aldir(i,ii)
+        asdifavg(i) = asdifavg(i)+cam_in%asdif(i,ii)
+        aldifavg(i) = aldifavg(i)+cam_in%aldif(i,ii)
+      enddo
+      lhfavg(i) = lhfavg(i)/float(num_inst_atm)
+      shfavg(i) = shfavg(i)/float(num_inst_atm)
+      wsxavg(i) = wsxavg(i)/float(num_inst_atm)
+      wsyavg(i) = wsyavg(i)/float(num_inst_atm)
+            snowhlandavg(i) = snowhlandavg(i)/float(num_inst_atm)
```

Comment:
Fix indentation

Reply:
I will check through all changed files and fix them all